client/js/src/client.ts

```diff
@@ -1014,6 +1053,12 @@ export function api_factory(
 				event_stream = new EventSource(url);
 				event_stream.onmessage = async function (event) {
 					let _data = JSON.parse(event.data);
+					if (_data.session_hash) {
+						await Promise.all(
+							event_ids.map((event_id) => event_callbacks[event_id](_data))
```

Comment:
If this is the only reason we're keeping track of event_ids, we already have that list in the keys of event_callbacks. No need to create a second list right?

Reply:
also would prefer if the logic for targetting every event listener was more explicit than if the session_hash key is present. We don't use the session_hash value anyway. I think it'd make more sense that if an event_id is not provided, we trigger every event.

Reply:
Done!

Reply:
Also made the change about checking the absence of event_id to trigger all listeners!